### ISSUE TITLE
noVNC is not working for VMs with password enabled

### ISSUE THREAD

**Issue by kvaps** [OpenNebula/one#3597]
**Description**
After upgrade to 5.8.4 I faced with the issue when most of VMs have no working VNC. Usually it is stacks on this screen without any errors:
![image](https://user-images.githubusercontent.com/7556217/63196383-764d6100-c075-11e9-968d-e335d040fb4d.png)

But after few minutes, it shows:
![image](https://user-images.githubusercontent.com/7556217/63196557-d2b08080-c075-11e9-9bad-f8d4db8ac044.png)

and error in browser's console:
```
Failed when connecting: Connection closed (code: 1006) rfb.js:677
    _fail rfb.js:677
    RFB rfb.js:277
    onclose websock.js:227
```

**To Reproduce**

- Create VM with 
  ```
  GRAPHICS = [
    LISTEN = "0.0.0.0",
    RANDOM_PASSWD = "YES",
    TYPE = "VNC" ]
  ```
- Try to connect via VNC

**Expected behavior**
Connected VNC

**Details**
 - Affected Component: Sunstone
 - Hypervisor: KVM
 - Version: 5.8.4

**Additional context**

After rollback to 5.8.1 vnc is working again

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by alitvak69** [comment]
I have noticed some slowness, whence a black bar in the beginning.   However, it loads the console in a few seconds.

**Comment by kvaps** [comment]
Looks like novnc client problem, I have sunstone and novnc in separated containers.

So I can confirm that problem not in novnc-server but in client, I've tested few scenarios:

* Sunstone 5.8.1 and novnc server from 5.8.4 - working fine
* Sunstone 5.8.4 and novnc server from 5.8.4 - can't connect to some VMs

I also made dump of packets on the host with the both VMs to see what is actually wrong there:

Left side - bad case, Right side - good case
![image](https://user-images.githubusercontent.com/7556217/63212614-7f870e00-c107-11e9-9820-4ec7a6284eea.png)

What did I found, actually nothing, just see that after 10th packets novnc-server does not sends any packets.

**Comment by kvaps** [comment]
Got it! Problem occurs only for VMs with enabled password authentication for VNC

**Comment by kvaps** [comment]
probably it is stopped working after https://github.com/OpenNebula/one/issues/2346

**Comment by tinova** [comment]
thanks kvaps and tryfunc!

**Comment by goncaloist** [comment]
Hello. I have been testing opennebula 5.8.5 and noticed that apparently this issue was not completely resolved with the fix.

The VNC windows is now available when using a password, but is still not displayed when the option "Open in a new Windows" is then used. 

I also tried inserting the fix manually and rebuilding the 5.8.4 packages, with the same result.  Hope that this issue was the right place in which to pass this information (reopening the issue?).

**Comment by jloboescalona2** [comment]
hi, @goncaloist, when you click on the button in the url do these parameters come out? a &encrypt=no&title=TEST!2&password=PASSWORD

**Comment by kvaps** [comment]
I've install 5.8.5 and found that now password is working for VNC for me

**Comment by kvaps** [comment]
But as @goncaloist said ![b](https://user-images.githubusercontent.com/7556217/66311044-d641e480-e90d-11e9-8e2f-c32daa7e3c01.png) button is not working anymore:

![image](https://user-images.githubusercontent.com/7556217/66310974-b3173500-e90d-11e9-89fa-84b7a53013fe.png)

**Comment by kvaps** [comment]
Hm, it seems after restart sunstone and memcache is started working as it should

**Comment by goncaloist** [comment]
> hi, @goncaloist, when you click on the "Open in a new Windows" button in the url do these parameters come out? a &encrypt=no&title=TITLEVM&password=PASSWORD

Hello. I was seeing only: "&encrypt=yes&title=TITLEVM". Now tried adding the &password=... and it worked. 
I noticed that my "vnc_request_password" option was set to true. Changed it back to false and it now opens the new window. I had misunderstood the purpose of that option. It was no bug, working as intended.
Sorry, I had stumbled in this bug with 5.8.4 previously and though my "open in new window" issue was something related.
